src/AcceptanceTests/Receiving/When_receiving_a_message.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace NServiceBus.Transport.AzureServiceBus.AcceptanceTests.Sending.Receiving
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+    using Pipeline;
+
+    public class When_receiving_a_message
+    {
+        [Test]
+        public async Task Should_have_access_to_brokered_message_system_properties_via_the_context_bag()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.When(
+                    (session, c) => session.SendLocal(new Message())))
+                .Done(c => c.LockedUntilUtcFromHandler == c.LockedUntilUtcFromBehavior)
+                .Run();
+        }
+
+        public class Context : ScenarioContext
+        {
+            public DateTime LockedUntilUtcFromHandler { get; set; }
+            public DateTime LockedUntilUtcFromBehavior { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(c => c.Pipeline.Register(b => new CheckContextForValidUntilUtc(b.Build<Context>()), "Behavior to validate context bag contains brokered message system properties"));
+            }
+
+            public class Handler : IHandleMessages<Message>
+            {
+                public Context TestContext { get; set; }
+
+                public Task Handle(Message request, IMessageHandlerContext context)
+                {
+                    TestContext.LockedUntilUtcFromHandler = context.Extensions.Get<Microsoft.Azure.ServiceBus.Message.SystemPropertiesCollection>().LockedUntilUtc;
```

Comment:
I wasn't aware that this is a nested class. I checked now again the ` Message` type and I'm thinking we could really just expose the native message directly if we wanted to. Yes you could then technically do dump stuff with the body but it seems to unlock quite a few interesting native scenarios people can build by having the full power of message available. I remember talking to Brandon and Andreas when they built the ` BasicDeliveryEventArgs`  exposure in RabbitMQ and the conclusion there was that while users can do dangerous stuff it is an advanced low level seam that requires a good understanding of the rabbitmq SDK anyway. I see how we could make the same tradeoffs here